app/models/concerns/geocodeable.rb

```diff
@@ -0,0 +1,34 @@
+# Add to a model that can be geolocated
+# Expects `latitude` and `longitude` columns to be defined.
+module Geocodeable
+  extend ActiveSupport::Concern
+
+  included do
+    geocoded_by :geocode_data
+    after_validation :geocode, if: :should_be_geocoded?
+
+    attr_accessor :skip_geocoding
+
+    def skip_geocoding?
+      !!skip_geocoding
+    end
+
+    # Override in Geocodeable model.
+    def geocode_data; end
```

Comment:
Would it make more sense to default to `@geocode_data ||= address`, since most of the models do that?

Reply:
Good catch thanks -- updated in 0ab79e2e